openshift-201/rh201-pod-auto-scale.md

```diff
@@ -0,0 +1,215 @@
+# Autoscaling Pods
+## Objectives
+
+After completing this section you should have a better understanding of Horizontal Pod Autoscalers (HPA), Vertical Pod Autoscalers (VPA), and Pod Disruption Budgets (PDB). And go through the steps to configure and observe these resources.
+
+## Horizontal Pod Autoscalers
+
+OpenShift can autoscale a deployment based on current load on the application pods, by means of a HorizontalPodAutoscaler resource type.
+
+A horizontal pod autoscaler (HPA) resource uses performance metrics collected by the OpenShift Metrics subsystem. The Metrics subsystem comes pre-installed in OpenShift 4.
+
+You can also create an HPA for DeploymentConfig, ReplicaSet, ReplicationController, or StatefulSet objects.
+
+This command will create a horizontal pod autoscaler resource that changes the number of replicas on the hello-world-nginx deployment to keep its pods under 80% of their total requested CPU usage.
+
+```
+oc autoscale deployment/hello-world-nginx --min 1 --max 5 --cpu-percent 80
+```
+
+The maximum and minimum values for the horizontal pod autoscaler resource serve to accommodate bursts of load and avoid overloading the OpenShift cluster. If the load on the application changes too quickly, then it might be advisable to keep a number of spare pods to cope with sudden bursts of user requests. Conversely, too many pods can use up all cluster capacity and impact other applications sharing the same OpenShift cluster.
+
+Edit the load-test deployment requests environment variable which will re-trigger the deployment to start a load-test pod that will send traffic to the hello-world-nginx pod. You should see the number of pods increase as the CPU metrics grow.
+
+To get information about horizontal pod autoscaler resources in the current project, use the oc get command.
+
+```
+oc get hpa 
+NAME                REFERENCE                      TARGETS    MINPODS   MAXPODS   REPLICAS
+hello-world-nginx   Deployment/hello-world-nginx   600%/80%   1         5          4        
+```
+
+The horizontal pod autoscaler initially has a value of `<unknown>` in the TARGETS column. It might take up to five minutes before <unknown> changes to display a percentage for current usage.
+
+A persistent value of `<unknown>` in the TARGETS column might indicate that the deployment does not define resource requests for the metric. The horizontal pod autoscaler will not scale these pods.
```

Comment:
would you also explain what the `600%/80%` value in targets mean?

Reply:
updated!